paypro-gateways-woocommerce/includes/paypro/wc/gateway.php

```diff
@@ -1,18 +1,24 @@
 <?php if(!defined('ABSPATH')) exit; // Exit if accessed directly
 
-abstract class PayPro_WC_Gateway_Abstract extends WC_Payment_Gateway
+class PayPro_WC_Gateway extends WC_Payment_Gateway
 {
-    protected $default_title;
+	/**
+	 * panther restapi responses
+	 */
+	protected const PAYPROAPIRES_NOTSUBSCRIPTED = "Not subscribed to money transfer service";
 
-    protected $default_description;
+    protected $default_title;
 
     protected $default_logo;
 
     protected $issuer;
+    protected $gateway_title = '';
+    protected $gateway_description = '';
+    public $has_fields = FALSE;
```

Comment:
Does this need to be `public`?

Reply:
![image](https://user-images.githubusercontent.com/55430516/142640506-f7022dff-f191-4ddb-8a34-786ff2f5b4b3.png)
wordpress will also throw a critical error

![image](https://user-images.githubusercontent.com/55430516/142640602-52c193d5-1dec-4bdd-9367-f5da0d29db12.png)

Reply:
I see, it is a `public` attribute https://woocommerce.wp-a2z.org/oik_api/wc_payment_gatewayhas_fields/